fv3gfs-util/tests/test_partitioner.py

```diff
@@ -108,150 +107,436 @@ def test_tile_extent_from_rank_metadata(array_extent, array_dims, layout, tile_e
+            (5, 8, 8),
+            (3, 3),
+            1,
+            (slice(0, 5), slice(0, 1), slice(1, 7)),
+            False,
+            float(1 / 6),
+            id="54_rank_botmid_right_no_overlap_square_layout_sixth_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            4,
+            (slice(0, 5), slice(1, 7), slice(1, 7)),
+            False,
+            float(1 / 6),
+            id="54_rank_midmid_right_no_overlap_square_layout_sixth_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            5,
+            (slice(0, 5), slice(1, 7), slice(7, 8)),
+            False,
+            float(1 / 6),
+            id="54_rank_midright_right_no_overlap_square_layout_sixth_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            8,
+            (slice(0, 5), slice(7, 8), slice(7, 8)),
+            False,
+            float(1 / 6),
+            id="54_rank_topright_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            0,
+            (slice(0, 5), slice(0, 2), slice(0, 2)),
+            False,
+            float(1 / 3),
+            id="96_rank_farbotfarleft_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            1,
+            (slice(0, 5), slice(0, 2), slice(2, 8)),
+            False,
+            float(1 / 3),
+            id="96_rank_farbotcloseleft_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            6,
+            (slice(0, 5), slice(2, 8), slice(8, 14)),
+            False,
+            float(1 / 3),
+            id="96_rank_closebotcloseright_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 16, 16),
+            (4, 4),
+            14,
+            (slice(0, 5), slice(14, 16), slice(8, 14)),
+            False,
+            float(1 / 3),
+            id="96_rank_fartopcloseright_right_no_overlap_square_layout_third_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Y_INTERFACE_DIM, fv3gfs.util.X_INTERFACE_DIM],
```

Comment:
The case we usually see for interfaces is that one dimension is an interface dimension and the other is not - could you add one or two cases like this?

Reply:
Test cases have been added for both variants (just X or Y interfaces).